buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -293,21 +295,25 @@ public ElasticsearchNode singleNode() {
         return getFirstNode();
     }
 
-    private void addWaitForUri(String description, String uri) {
-        waitConditions.put(description, (node) -> {
+    private void addWaitForClusterHealth() {
```

Comment:
Now that this is just a generic health wait, is there any scenario in which we would _not_ eventually add this wait condition? If so we should just initialize `waitConditions` with this in the constructor or something. I don't think `waitForConditions()` should do this as we may call that method several times and end up adding this same condition multiple times.

Reply:
good catch! Fixed.